lib/common_test/src/ct_ssh.erl

```diff
@@ -367,125 +392,188 @@ receive_response(SSH, ChannelId, Timeout) when is_integer(Timeout) ->
 %%%      <code>false</code> to wait for more data from the server.
 %%%      (Note that even if a fun is supplied, the function returns
 %%%      immediately if the server closes the channel).</p>
+%%% @end
+-spec receive_response(SSH, ChannelId, End, Timeout) ->
+                              {ok,Data} | {timeout,Data} | {error,Reason} when
+      SSH :: connection(),
+      ChannelId :: integer(),
+      End :: Fun | close | timeout,
+      Fun :: fun((ssh_connection:ssh_event_msg()) -> boolean()),
```

Comment:
From the code, you can see the fun is called in <code>do_recv_response/5</code>, and the input is the fourth element of <code>data_events()</code>, which is a <code>binary()</code>. As far as I can see, there is no place where this fun is called, so I think it is safe to specify <code>binary()</code> here.